src/util.cpp

```diff
@@ -929,6 +929,14 @@ bool hasNonPublicParam(const CXXMethodDecl *decl)
         {
             CXXRecordDecl *cxx_record_decl
                 = cast<CXXRecordDecl>(record_type->getDecl());
+
+            // Allow the parent class, in which the method is defined. In this
+            // case, the access specifier is As_none.
+            if (decl->getParent() == cxx_record_decl)
+            {
+                continue;
+            }
+
```

Comment:
Hmm, I am not sure why this needed to be modified?

Reply:
The access type is `AS_none` for a parameter type in a method whose parent class is the parameter type itself. Without this block, we ignore all the methods that take a type of their parent class type, which I don't think what we want. For example,
```cpp
class Vector3d {
public:
  Vector3d operator+(const Vector3d& other);  // This method is ignored because `cxx_record_decl->getAccess() == AS_none` in this case
};
```

Reply:
Ah ok.  C++ is full of so many quirks 😓 

The documentation helpfully says:
https://clang.llvm.org/doxygen/Specifiers_8h_source.html#l00117
> A C++ access specifier (public, private, protected), plus the
> special value "none" **which means different things in different contexts.**

🤦

Reply:
Yeah, it should have been named with something like `AS_different_things_in_different_contexts` so that the user won't underestimate it what it can mean. :joy: